app/routes.py

```diff
@@ -1 +1,237 @@
+    request_body = request.get_json()
+
+    try:
+        new_task = Task(title=request_body["title"], description=request_body["description"])
+            
+    except Exception as e:
+        return jsonify({"details": "Invalid data"}), 400
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": new_task.to_dict()}), 201
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task_by_id(task_id):
+    '''Get one task by id, validates id before executing query.'''
+    task = validate_model_by_id(Task, task_id)
+
+    return jsonify({"task": task.to_dict()}), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task_by_id(task_id):
+    '''Update a task by id, id validated before executing request.'''
+    request_body = request.get_json()
+    task = validate_model_by_id(Task, task_id)
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return jsonify({"task": task.to_dict()}), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task_by_id(task_id):
+    '''Delete a task by id, id validated before executing request.'''
+    task = validate_model_by_id(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return jsonify({"details": f'Task {task_id} "{task.title}" successfully deleted'}), 200
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    '''Updating task to completed using datetime function. Task ID validated before executing request.
+    Sends message of completion to slack channel task-notifications.'''
+    
+    task = validate_model_by_id(Task, task_id)
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    # slack api call to post to task-notifications
+    slack_oauth_token = os.environ.get("SLACK_OAUTH_TOKEN")
+    url = "https://slack.com/api/chat.postMessage"
+    params = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.title}"
+    }
+    headers = {"Authorization": f"Bearer {slack_oauth_token}"}
+
+    requests.patch(url=url, params=params, headers=headers)
+
+    return jsonify({"task": task.to_dict()}), 200
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    '''Marks a task as incomplete by id. ID validated before request.'''
+    task = validate_model_by_id(Task, task_id)
+
+    task.completed_at = None
+    db.session.commit()
+
+    return jsonify({"task": task.to_dict()}), 200
+
+
+####################Goal routes###################
+
+@goals_bp.route("", methods=["POST"])
+def create_one_goal():
+    '''Creates a new goal, requires title, id will be autogenerated.'''
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except Exception as e:
+        return jsonify({"details": "Invalid data"}), 400
```

Comment:
Similar to the last post request, having some way to let the user know what was wrong would be helpful.